src/Faker/Provider/es_MX/Person.php

```diff
@@ -0,0 +1,466 @@
+    public static function personRfc($firstName = null, $lastNameFather = null, $lastNameMother = null, $birthDate = null, $gender = null)
+    {
+        $gender = in_array($gender, array(Person::GENDER_MALE, Person::GENDER_FEMALE)) ? $gender : static::randomElement(array(Person::GENDER_MALE, Person::GENDER_FEMALE));
+
+        if ($gender === Person::GENDER_MALE) {
+            $firstName = self::removeAccents(mb_strtoupper($firstName ? $firstName : static::firstNameMale()));
+        } else {
+            $firstName = self::removeAccents(mb_strtoupper($firstName ? $firstName : static::firstNameFemale()));
+        }
+
+        $lastNameFather = self::removeAccents(mb_strtoupper($lastNameFather ? $lastNameFather : static::randomElement(static::$lastNames)));
+        $lastNameMother = self::removeAccents(mb_strtoupper($lastNameMother ? $lastNameMother : static::randomElement(static::$lastNames)));
+        $birthDate = $birthDate ? $birthDate : \Faker\Provider\DateTime::dateTimeBetween();
+
+        $rfc = self::commonPart('rfc', $firstName, $lastNameFather, $lastNameMother, $birthDate);
+        $rfc .= strtoupper(static::lexify('???'));
+
+        return $rfc;
+    }
+
+    /**
+     * HELPERS
+     */
+
+    protected static function commonPart($type, $firstName = null, $lastNameFather = null, $lastNameMother = null, $birthDate = null)
+    {
+        $common  = substr($lastNameFather, 0, 1);
+        $common .= self::firstInternalVowel($lastNameFather);
+        $common .= $lastNameMother ? substr($lastNameMother, 0, 1) : 'X';
+        $common .= substr($firstName, 0, 1);
+        if ($type == 'curp') {
+            $common = str_replace(array_keys(self::$badWordsCURP), array_values(self::$badWordsCURP), $common);
+        } else {
+            $common = str_replace(array_keys(self::$badWordsRFC), array_values(self::$badWordsRFC), $common);
+        }
+        $common .=$birthDate->format('ymd');
+        return $common;
+    }
+
+    protected static function firstInternalVowel($word)
+    {
+        if (preg_match('/([AEIOU])/', substr($word, 1), $matches)) {
+            return $matches[1];
+        }
+        return 'X';
+    }
+
+    protected static function secondInternalConsonant($word)
+    {
+        if (preg_match('/.[AEIOU]*([BCDFGHJKLMNPQRSTUVWXYZ]{1})/', $word, $matches)) {
+            return $matches[1];
+        }
+        return 'X';
+    }
+
+    protected static function removeCommonNames($name)
+    {
+        foreach (self::$notAcceptedNames as $notAcceptedName) {
+            $name = preg_replace($notAcceptedName, '', $name);
+        }
+        return $name;
+    }
+    protected static function removePrefixes($name)
+    {
+        foreach (self::$prefixes as $prefix) {
+            $name = preg_replace($prefix, '', $name);
+        }
+        return $name;
+    }
+    protected static function removeAccents($name)
+    {
+        return str_replace(array_keys(self::$accents), array_values(self::$accents), $name);
```

Comment:
Not sure this one is utf8-safe, can you add unit tests?

Reply:
there is no `mb_str_replace` because `str_replace` contrary to belief works perfectly on multibyte characters